src/uwtools/drivers/driver.py

```diff
@@ -16,103 +11,70 @@
-    forecast. Subsequent methods will be used to stage the input files and run the forecast.
-
-    Attributes
-    ----------
-    config_path : Path
-        The file path to the configuration file to be parsed.
-
-    Methods
-    -------
-    requirements()
-        Recursively parse config and platform files to determine and
-        fill in any dependencies.
-
-    resources()
-        Determine necessary task objects and fill in resource requirements of each.
-        Returns a Config object containing the HPC resources needed.
-
-    validate()
-        Validates the objects generated by the driver from the provided
-        config and platform files. If the --dry-run flag is provided, complete all
-        stages through validation and print results without running the forecast.
-
-
-    output()
-        Holds the knowledge for how to modify a list of output files and
-        stages them in the working directory.
-
-    job_card()
-        Turns the resources config object into a batch card for the
-        configured Task. Can be called from the command line on the front end
-        node to show the user what the job card would have looked like.
-
-    Notes
-    -------
-    Several functions such as create_model_config() and run() are relegated to
-    specific forecast drivers. This is because they are unique to the app being
-    run and will appropriately parse subsequent stages of the workflow.
+    An abstract class representing drivers for various NWP tools.
     """
 
-    def __init__(self, config_file: Optional[str] = None, log: Optional[Logger] = None):
+    def __init__(self, config_file: str, log: Optional[Logger] = None):
         """
-        Initialize the Forecast driver.
+        Initialize the driver.
         """
         self.log = log if log is not None else Logger()
-        if config_file is not None:
-            self.config_file = config_file
-            self._validate()
-        else:
-            self.log.info("No config file provided, available functions are limited.")
+        self._config_file = config_file
+        self._validate()
 
-    def _validate(self) -> bool:
+    # Public methods
+
+    @abstractmethod
+    def batch_script(self) -> None:
         """
-        Validates the objects supplied to the driver from the provided config and platform files.
+        Create a script for submission to the batch system.
         """
-        return config_validator.config_is_valid(
-            config_file=self.config_file,
-            validation_schema=self.schema,
-            log=self.log,
-        )
 
-    @property
     @abstractmethod
-    def schema(self):
+    def output(self) -> None:
         """
-        The schema to validate the config file against.
+        ???
         """
 
     @abstractmethod
-    def requirements(self):
+    def requirements(self) -> None:
         """
-        Recursively parse config and platform files to determine and fill in any dependencies.
+        ???
```

Comment:
I wasn't able to make sense of some of the existing docstrings where there's no implementation yet, so I blanked these out to be completed later. In general, docstrings should describe what a method does at a high level, without delving into implementation specifics. For example, a user running `help(fcstobj.requirements)` wouldn't need to know that recursion is being used -- they need to know what the method needs as input, what it returns, and what side effects it might have; they don't need to know _how_ the work gets done. `#` comments in code are the preferred place for algorithmic details.

I also added `-> None` return-type annotations, though these will probably need to be updated when the methods are actually implemented.

Reply:
Thats fair - I had put text here to effectively propose the planned structure as part of the facade, though its not needed now, not to mention depreciated